### PR TITLE
WIP: add feature: plane timestamps

### PR DESCRIPTION
**Pull request by kd0aij**
@Arne-W This is a small mod to your bugfix branch to add GPS time to plane placemarks.
I had to robustify the GPS record validation check because I was getting records with zero week and week_msec fields, and that screws up Google Earth.
If it's working correctly, you should get a "time slider" at the upper left of the GE map display as soon as you load a KMZ file.

This could easily be added to the flight path data also.

### CODE REVIEW

File: src/output/kmlcreator.cc [hunk 1]
```diff
@@ -491,7 +491,32 @@ void KMLCreator::writePlanePlacemarkElement(QXmlStreamWriter &writer, Placemark
 
     int index = 0;
     foreach(GPSRecord c, p->mPoints) {
+        std::string week_ms_str = c.msec().toStdString();
+        const char *tschar = week_ms_str.c_str();
+        char* endptr;
+        uint32_t week_ms = strtoul(tschar, &endptr, 10);
```
Comment (Arne-W) [inline src/output/kmlcreator.cc]: fetching the week_ms can be done much easier: `int weekMS = c.msec().toInt();` for example.

File: src/output/kmlcreator.cc [hunk 2]
```diff
@@ -491,7 +491,32 @@ void KMLCreator::writePlanePlacemarkElement(QXmlStreamWriter &writer, Placemark
...
+
+        bool status;
```
Comment (Arne-W) [inline src/output/kmlcreator.cc]: the return value is not processed - remove it please

**Comment by kd0aij** [comment]
That code is a quick & dirty hack just to see if the timestamps would work as expected, and I assumed week_ms was unsigned for a good reason.
I also assume there's a utility around somewhere to do a proper conversion from GPS iTOW to UTC time, without the need for reinventing one. Need to find that next.

**Comment by Arne-W** [comment]
Ah - all right :+1:

**Pull request closed without merging**